src/coreclr/inc/crosscomp.h

```diff
@@ -705,77 +714,38 @@ typedef struct _KNONVOLATILE_CONTEXT_POINTERS_EX
 
 #endif
 
-#if defined(DACCESS_COMPILE) && defined(TARGET_UNIX)
-// This is a TARGET oriented copy of CRITICAL_SECTION and PAL_CS_NATIVE_DATA_SIZE
-// It is configured based on TARGET configuration rather than HOST configuration
-// There is validation code in src/coreclr/vm/crst.cpp to keep these from
-// getting out of sync
-
-#define T_CRITICAL_SECTION_VALIDATION_MESSAGE "T_CRITICAL_SECTION validation failed. It is not in sync with CRITICAL_SECTION"
-
-#if defined(TARGET_OSX) && defined(TARGET_X86)
-#define DAC_CS_NATIVE_DATA_SIZE 76
-#elif defined(TARGET_APPLE) && defined(TARGET_AMD64)
-#define DAC_CS_NATIVE_DATA_SIZE 120
-#elif defined(TARGET_APPLE) && defined(TARGET_ARM64)
-#define DAC_CS_NATIVE_DATA_SIZE 120
-#elif defined(TARGET_FREEBSD) && defined(TARGET_X86)
-#define DAC_CS_NATIVE_DATA_SIZE 12
-#elif defined(TARGET_FREEBSD) && defined(TARGET_AMD64)
-#define DAC_CS_NATIVE_DATA_SIZE 24
-#elif defined(TARGET_FREEBSD) && defined(TARGET_ARM64)
-#define DAC_CS_NATIVE_DATA_SIZE 24
-#elif (defined(TARGET_LINUX) || defined(TARGET_ANDROID)) && defined(TARGET_ARM)
-#define DAC_CS_NATIVE_DATA_SIZE 80
-#elif (defined(TARGET_LINUX) || defined(TARGET_ANDROID)) && defined(TARGET_ARM64)
-#define DAC_CS_NATIVE_DATA_SIZE 104
-#elif defined(TARGET_LINUX) && defined(TARGET_LOONGARCH64)
-#define DAC_CS_NATIVE_DATA_SIZE 96
-#elif (defined(TARGET_LINUX) || defined(TARGET_ANDROID)) && defined(TARGET_X86)
-#define DAC_CS_NATIVE_DATA_SIZE 76
-#elif (defined(TARGET_LINUX) || defined(TARGET_ANDROID)) && defined(TARGET_AMD64)
-#define DAC_CS_NATIVE_DATA_SIZE 96
-#elif defined(TARGET_LINUX) && defined(TARGET_S390X)
-#define DAC_CS_NATIVE_DATA_SIZE 96
-#elif defined(TARGET_LINUX) && defined(TARGET_RISCV64)
-#define DAC_CS_NATIVE_DATA_SIZE 96
-#elif defined(TARGET_LINUX) && defined(TARGET_POWERPC64)
-#define DAC_CS_NATIVE_DATA_SIZE 96
-#elif defined(TARGET_NETBSD) && defined(TARGET_AMD64)
-#define DAC_CS_NATIVE_DATA_SIZE 96
-#elif defined(TARGET_NETBSD) && defined(TARGET_ARM)
-#define DAC_CS_NATIVE_DATA_SIZE 56
-#elif defined(TARGET_NETBSD) && defined(TARGET_X86)
-#define DAC_CS_NATIVE_DATA_SIZE 56
-#elif defined(__sun) && defined(TARGET_AMD64)
-#define DAC_CS_NATIVE_DATA_SIZE 48
-#elif defined(TARGET_HAIKU) && defined(TARGET_AMD64)
-#define DAC_CS_NATIVE_DATA_SIZE 56
-#elif defined(TARGET_WASM)
-#define DAC_CS_NATIVE_DATA_SIZE 76
+#if defined(TARGET_APPLE)
+#define DAC_CS_MAX_SIZE 96
+#elif defined(TARGET_FREEBSD)
+#define DAC_CS_MAX_SIZE 16
+#elif defined(TARGET_LINUX) || defined(TARGET_ANDROID)
+#define DAC_CS_MAX_SIZE 64
+#elif defined(TARGET_WINDOWS)
+#define DAC_CS_MAX_SIZE 40
```

Comment:
```suggestion
:#ifdef TARGET_64BIT
#define DAC_CS_MAX_SIZE 40
#else
#define DAC_CS_MAX_SIZE 24
#endif
```
Avoid regressing wn-x86